cy43439.go

```diff
@@ -152,162 +151,6 @@ func NewDevice(spi drivers.SPI, cs, wlRegOn, irq, sharedSD machine.Pin) *Device
 	return d
 }
 
-// EnableStaMode enables the wifi interface in station mode.
-//
-// ref: void cyw43_arch_enable_sta_mode()
-// ref: void cyw43_wifi_set_up(cyw43_t *self, int itf, bool up, uint32_t country)
-func (d *Device) EnableStaMode(country uint32) error {
-	return d.wifiSetup(whd.CYW43_ITF_STA, true, country)
-}
-
-// ref: int cyw43_arch_wifi_connect_timeout_ms(const char *ssid, const char *pw, uint32_t auth, uint32_t timeout_ms)
-func (d *Device) WifiConnectTimeout(ssid, pass string, auth uint32, timeout time.Duration) error {
-	deadline := time.Now().Add(timeout)
-
-	if err := d.wifiConnect(ssid, pass, auth); err != nil {
-		return err
-	}
-	status := int32(whd.CYW43_LINK_UP + 1)
-	for status >= 0 && status != whd.CYW43_LINK_UP {
-		newStatus := d.tcpipLinkStatus(whd.CYW43_ITF_STA)
-		if newStatus == whd.CYW43_LINK_NONET {
-			newStatus = whd.CYW43_LINK_JOIN
-			if err := d.wifiConnect(ssid, pass, auth); err != nil {
-				return err
-			}
-		}
-		if newStatus != status {
-			d.info("WifiConnectTimeout:status_change", slog.Int("newstatus", int(newStatus)), slog.Int("oldstatus", int(status)))
-			status = newStatus
-		}
-		if time.Until(deadline) <= 0 {
-			return netlink.ErrConnectTimeout
-		}
-		time.Sleep(1 * time.Second)
-	}
-	if status == whd.CYW43_LINK_UP {
-		return nil
-	} else if status == whd.CYW43_LINK_BADAUTH {
-		return netlink.ErrAuthFailure
-	}
-	return netlink.ErrConnectFailed
-}
-
-// ref: cyw43_wifi_set_up(cyw43_t *self, int itf, bool up, uint32_t country)
-func (d *Device) wifiSetup(itf uint8, up bool, country uint32) (err error) {
-	d.lock()
-	defer d.unlock()
-	if !up {
-		if itf == itfAP {
-			err = d.wifiAPSetUp(false)
-		}
-		return err
-	}
-
-	if d.itfState == 0 {
-		if err := d.wifiOn(country); err != nil {
-			return err
-		}
-		if err := d.wifiPM(defaultPM); err != nil {
-			return err
-		}
-	}
-
-	if itf == whd.CYW43_ITF_AP {
-		if err = d.wifiAPInit(); err != nil {
-			return err
-		}
-		if err = d.wifiAPSetUp(true); err != nil {
-			return err
-		}
-	}
-	// itf == AP:cyw43_wifi_ap_init(self);  cyw43_wifi_ap_set_up(self, true);
-	if d.itfState&(1<<itf) == 0 {
-		// Reinitialize tcpip here.
-		d.itfState |= 1 << itf
-	}
-	return nil
-}
-
-func (d *Device) isSTAActive() bool { return (d.itfState>>itfSTA)&1 != 0 }
-func (d *Device) isAPActive() bool  { return (d.itfState>>itfAP)&1 != 0 }
```

Comment:
FYI this was the bug, the enum had itfSTA be 1 when in the reference it was 0.

Reply:
Yikes!  Good catch.  I guess they're trying to save memory in the device structure, but would it have been so bad to use an enum for mode (STA, AP, etc)?